### ISSUE TITLE
nerdtree: no icons on newly created files

### ISSUE THREAD

**Issue by dflupu** [ryanoasis/vim-devicons#153]
Same as issue #33 which is closed.
I'm using the latest versions of nerdtree and vim-devicons.

#### Requirements

If you are **not** sure whether or not what you are reporting is actually an issue **or** you have a general question please message in the [Gitter Chat Room](https://gitter.im/ryanoasis/vim-devicons)

These items you acknowledge and mark off with 'x' please

- [x] I have searched the [issues](https://github.com/ryanoasis/vim-devicons/issues) for my issue and found nothing related and/or helpful
- [x] I have read or at least glanced at the [FAQ](https://github.com/ryanoasis/vim-devicons#faq--troubleshooting)
- [x] I have read or at least glanced at the [Wiki](https://github.com/ryanoasis/vim-devicons/wiki)

These items you must provide answers to. Make sure to add **all the information needed to understand the bug** so that someone can help. If the info is missing we'll add the 'Needs more information' label and _may_ choose to close the issue until there is enough information.

- [x] What **version** of vim are you using?
  - hint: `vim --version`

NVIM v0.1.4-5-g5330aa1
and
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Apr  8 2016 14:36:27)
Included patches: 1-1718
- [x] What **version** of vim-devicons are you using?
  - hint: `:echo webdevicons#version()`

 0.8.2
- [x] Are you using vim from the terminal or a GUI vim? terminal
- [x] Are you using Mac, Linux or Windows? linux

#### Optional

- [ ] Provide **screenshots** where appropriate

**Comment by ryanoasis** [comment]
Thanks ill try to replicate today

**Comment by ryanoasis** [comment]
So far I am unable to replicate so far though I have not tried the exact setup you have.

Do you have a link to your `vimrc` you are using so I can test if there is anything specific in it?

**Comment by dflupu** [comment]
I would have a hard time sharing my vimrc, since it's split into multiple files.
I went ahead and tested with a vimrc containing only the following
```
call plug#begin()
Plug 'scrooloose/nerdtree'
Plug 'ryanoasis/vim-devicons'
call plug#end()
```
and the issue did not go away. Autoload directory only contains vim-plug. Also tested with pathogen.

**Comment by ryanoasis** [comment]
@dflupu Ah okay thanks, I'll try again

**Comment by ryanoasis** [comment]
can you tell me what you get for the output of `echo g:NERDTreeUpdateOnCursorHold` ?

**Comment by dflupu** [comment]
E121: Undefined variable: g:NERDTreeUpdateOnCursorHold                                                           
E15: Invalid expression: g:NERDTreeUpdateOnCursorHold

**Comment by ryanoasis** [comment]
Hey thanks for testing that. Looks like you found a bug (oversight on my part).

See I use [nerdtree-git-plugin](https://github.com/Xuyuanp/nerdtree-git-plugin) and I stole/borrowed a fix idea from there for refreshing the nerdtree flags but looks like I forgot to even set that variable!

See my commit on the initial attempt at a fix: https://github.com/ryanoasis/vim-devicons/commit/710f3b2af7721b6b331073792b67694ac9a3e942

But I need to do something (as shown here: https://github.com/Xuyuanp/nerdtree-git-plugin/blob/e71e62fb5ca3e49035a7f2a85c81e5b85b1cf138/nerdtree_plugin/git_status.vim#L37-L39) for setting that variable :blush: 

So basically I could **not** replicate your issue **until** I removed the [nerdtree-git-plugin](https://github.com/Xuyuanp/nerdtree-git-plugin) plugin and now I get the same error as you.

**Comment by ryanoasis** [comment]
The fix is up on [0.8.3](https://github.com/ryanoasis/vim-devicons/tree/0.8.3) branch if you would like to test it out.

I won't be able to do any changes until Tuesday

**Comment by dflupu** [comment]
I've tested it a bit. It does work, but it takes some time until the icon appears after a file is created. Seconds, even. Is it based on some kind of timeout ?

**Comment by ryanoasis** [comment]
@dflupu Yeah basically it is as far as I understand. I will see if I can try to improve the speed a bit.

Try to see what you have set for `updatetime`

For me I have it set to [250 for `gitgutter` in my `vimrc`](https://github.com/ryanoasis/dotfiles/blob/master/.vimrc#L489) so that is why I probably see the icon show up fairly quickly, without it the default seems to be 2 seconds (2000).

**Comment by ryanoasis** [comment]
Great! No problem :smile: 

I will try to add this info to the readme for the next release
